lib/osf-components/addon/components/move-file-modal/component.ts

```diff
@@ -0,0 +1,249 @@
+import Store from '@ember-data/store';
+import { assert } from '@ember/debug';
+import { inject as service } from '@ember/service';
+import { action, notifyPropertyChange } from '@ember/object';
+import { waitFor } from '@ember/test-waiters';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { allSettled, task, TaskInstance } from 'ember-concurrency';
+import { taskFor } from 'ember-concurrency-ts';
+import Intl from 'ember-intl/services/intl';
+import Toast from 'ember-toastr/services/toast';
+
+// import FileModel from 'ember-osf-web/models/file';
+import NodeModel from 'ember-osf-web/models/node';
+import { FileSortKey } from 'ember-osf-web/packages/files/file';
+import OsfStorageManager from 'osf-components/components/storage-provider-manager/osf-storage-manager/component';
+import OsfStorageFile from 'ember-osf-web/packages/files/osf-storage-file';
+import OsfStorageProviderFile from 'ember-osf-web/packages/files/osf-storage-provider-file';
+import CurrentUserService from 'ember-osf-web/services/current-user';
+// import RSVP from 'rsvp';
+
+interface MoveFileModalArgs {
+    filesToMove: OsfStorageFile[]; // TODO: type
+    manager: OsfStorageManager;
+}
+
+export default class MoveFileModalComponent extends Component<MoveFileModalArgs> {
+    @service store!: Store;
+    @service currentUser!: CurrentUserService;
+    @service intl!: Intl;
+    @service toast!: Toast;
+
+    @tracked currentNode!: NodeModel;
+    @tracked childNodeList: NodeModel[] = [];
+    @tracked childNodePage = 1;
+    @tracked totalChildNodes = 0;
+
+    @tracked startingFolder?: OsfStorageProviderFile | OsfStorageFile;
+    @tracked currentFolder?: OsfStorageProviderFile | OsfStorageFile; // TODO: types for other providers
```

Comment:
You can probably just use `ProviderFile | File` here instead of typing for each individually. And in other places.